src/polyhedralGravity/model/SplitNode.cpp

```diff
@@ -0,0 +1,108 @@
+            maybeFreeParam();//if both nodes are built, split parameters are no longer needed
+        }
+        return _lesser;
+    }
+
+    std::shared_ptr<TreeNode> SplitNode::getGreaterNode() {
+        if (!this->_greater) {                                                                 //node is not yet built
+            SplitParam childParam{*this->splitParam};                                          //copy parent param and modify to fit new node
+            childParam.boundingBox = KDTree::splitBox(this->_boundingBox, this->_plane).second;//get the greater box after splitting;
+            childParam.indexBoundFaces = *std::move(_triangleIndexLists[1]);                   //get the triangles of the greater box
+            childParam.splitDirection = static_cast<Direction>((static_cast<int>(this->splitParam->splitDirection) + 1) % DIMENSIONS);
+            _greater = TreeNodeFactory::treeNodeFactory(childParam);
+            maybeFreeParam();//if both nodes are built, split parameters are no longer needed
+        }
+        return _greater;
+    }
+
+    void SplitNode::maybeFreeParam() {
+        if (_lesser && _greater) {//both nodes are built
+            this->splitParam.reset();
+        }
+    }
+
+    void SplitNode::getFaceIntersections(const Array3 &origin, const Array3 &ray, std::set<Array3> &intersections) {
+        const auto delegates = getChildrenForIntersection(origin, ray);                                         //calculate affected child nodes
+        std::for_each(delegates.cbegin(), delegates.cend(), [&origin, &ray, &intersections](const auto &child) {//perform intersections on the child nodes
+            child->getFaceIntersections(origin, ray, intersections);
+        });
+    }
+
+    std::vector<std::shared_ptr<TreeNode>> SplitNode::getChildrenForIntersection(const Array3 &origin, const Array3 &ray) {
+        using namespace polyhedralGravity::util;
+        std::vector<std::shared_ptr<TreeNode>> delegates{};
+        delegates.reserve(2);                                          //a SplitNode has max two children, so no more space needed.
+        auto [t_enter, t_exit] = this->rayBoxIntersection(origin, ray);//calculate entry and exit points of the ray hitting the bounding box
+        if (t_exit < t_enter || t_exit < 0) {                          // bounding box was not hit because the ray passed the box or is moving into the opposite direction of it, TODO: consider moving to root node because only called there
+            return delegates;                                          //empty
+        }
+        const double t_split{rayPlaneIntersection(origin, ray)};                              //calculate point where plane was hit
+        if (!std::isinf(t_split) && t_split >= 0 && t_enter <= t_split && t_exit >= t_split) {//the split plane is hit inside of the bounding box -> both child boxes need to be checked //TODO: consider optimizing: t_param stay the same in sub boxes
```

Comment:
Can we simplify this condition somehow? Maybe as a metthod of the class `BoundingBox`.

The following two implementations are equivalent:
```c++
using Box = std::pair<Array3, Array3>
```
```c++
struct Box {
  Array3 first;
  Array3 second;
}
```

However, in the later one you can rename the members to something more expressive and add e.g. the comparison operator (And yes, you will still have the [Structured Binding](https://en.cppreference.com/w/cpp/language/structured_binding) available)


A good source for this is [godbolt](https://godbolt.org/z/bK3zv4a7o). As you can see when following the link, the assembly for both implementations is actually **the same**.